microbial_thermodynamics/cell_growth_ODE.py

```diff
@@ -112,6 +112,108 @@ def da(
     return uda
 
 
+# metabolic functions
+
+
+def calculate_kappa(
+    Gatp: float = 75,
+    G0: float = 1.5e5,
+    reaction_energy: float = 1 / 3,
+    R: float = 8.314,
+    T: float = 293.15,
+) -> float:
+    """Calculates the kappa constant."""
+    u = np.exp((-G0 - reaction_energy * Gatp) / R * T)
+    return u
+
+
+def calculate_theta(s: float, w: float) -> float:
+    """A thermodynamic factor that stops a reaction at equilibrium.
+
+    Args:
+     s: substrate concentration
+     w: waste product concentration
+     kappa: species equilibrium constant.
+    """
+    kappa = calculate_kappa()
+    u = (w / s) / kappa
+    return u
+
+
+def calculate_E(v: float, R: float, Q: float = 0.45, m: float = 1e8) -> float:
+    """Calculates enzyme copy number."""
+    u = m * (v - R - Q)
+    return u
+
+
+def q(
+    Gatp: float,
+    R: float,
+    s: float,
+    w: float,
+    ka: float = 10,
+    ks: float = 1.375e-3,
+    ra: float = 10,
+) -> float:
+    """Calculates reaction rate."""
+    E = calculate_E(1, R)
+    calculate_kappa(Gatp)
+    theta = calculate_theta(s, w)
+    u = (ka * E * s * (1 - theta)) / (ks + s * (1 + ra * theta))
+    return u
+
+
+def j(
+    Gatp: float, R: float, s: float, w: float, reaction_energy: float = 1 / 3
+) -> float:
+    """Calculates rate of ATP production in a species."""
+    u = reaction_energy * q(Gatp, R, s, w)
+    return u
+
+
+def calculate_produced(Gatp: float, R: float, s: float, w: float) -> float:
+    """Calculates rate a metabolite is produced per cell."""
+    u = q(Gatp, R, s, w)
+    return u
+
+
+def calculate_consumed(Gatp: float, R: float, s: float, w: float) -> float:
+    """Calculates rate a metabolite is consumed per cell."""
+    u = q(Gatp, R, s, w)
+    return u
+
+
+# below is the code i could not get working
+"""
+def calculate_sum_of_species(num_species: float = 2, N: float)-> float:
```

Comment:
Okay so what you really need here is an "if else" statement.

```python
if substrate_number == 1:
      consumed = calculate_consumed(R, s=c[0], w=[0])
      u = -consumed*N
else:
     produced = calculate_produced(R, s=c[0], w=[0])
     u = produced*N
return u
```

Reply:
That should work because what you are supplying the function is not a float but a vector of floats so `consumed` is a vector which then gets multiplied by a vector